sycl/include/sycl/ext/oneapi/experimental/chunk.hpp

```diff
@@ -122,30 +129,49 @@ template <size_t ChunkSize, typename ParentGroup> class chunk {
   }
 
 protected:
+  static constexpr size_t chunk_size = ChunkSize;
+
 #if defined(__SYCL_DEVICE_ONLY__) && defined(__NVPTX__)
   sub_group_mask Mask;
-#endif
 
-#if defined(__SYCL_DEVICE_ONLY__) && defined(__NVPTX__)
   chunk(ext::oneapi::sub_group_mask mask) : Mask(mask) {}
+
+  ext::oneapi::sub_group_mask getMask() const { return Mask; }
 #else
   chunk() {}
+
+#ifdef __SYCL_DEVICE_ONLY__
+  ext::oneapi::sub_group_mask getMask() const {
+    ext::oneapi::sub_group_mask::BitsType MaskBits{0};
+    MaskBits = ~MaskBits;
+    MaskBits >>= ext::oneapi::sub_group_mask::max_bits - ChunkSize;
+    MaskBits <<=
+        ext::oneapi::sub_group_mask::max_bits -
+        (((__spirv_SubgroupLocalInvocationId() / ChunkSize) + 1) * ChunkSize);
+    return sycl::detail::Builder::createSubGroupMask<
+        ext::oneapi::sub_group_mask>(MaskBits, __spirv_SubgroupMaxSize());
+  }
+#endif
 #endif
 
   friend chunk<ChunkSize, ParentGroup>
   chunked_partition<ChunkSize, ParentGroup>(ParentGroup parent);
 
   friend sub_group_mask sycl::detail::GetMask<chunk<ChunkSize, ParentGroup>>(
       chunk<ChunkSize, ParentGroup> Group);
+
+  template <size_t OtherChunkSize, typename OtherParentGroup>
+  friend class chunk;
 };
 
 // Chunked partition implementation
 template <size_t ChunkSize, typename ParentGroup>
-inline std::enable_if_t<sycl::is_group_v<std::decay_t<ParentGroup>> &&
-                            std::is_same_v<ParentGroup, sycl::sub_group>,
+inline std::enable_if_t<std::is_same_v<ParentGroup, sycl::sub_group> ||
+                            is_chunk_v<ParentGroup>,
                         chunk<ChunkSize, ParentGroup>>
-chunked_partition(ParentGroup parent) {
-  (void)parent;
+chunked_partition([[maybe_unused]] ParentGroup parent) {
```

Comment:
Just a comment: This `[[maybe_unused]]` is so much nicer than all the `(void)` casts!  Are there plans to make this change throughout the rest of the code?